tensorly/decomposition/_parafac2.py

```diff
@@ -95,6 +95,70 @@ def _project_tensor_slices(tensor_slices, projections):
     return tl.stack(slices)
 
 
+class _bro_thesis_line_search:
```

Comment:
```suggestion
class _BroThesisLineSearch:
```

Let's use the CapWords convention mentioned in [PEP 8](https://peps.python.org/pep-0008/#class-names).